### ISSUE TITLE
`gulp test.unit.js` sometimes fails to close the browser on failures

### ISSUE THREAD

**Issue by vicb** [angular/angular#1866]
On some failures, `gulp test.unit.js` fails to close the browser.

Then if you re-run the task, it will launch a new browser and the tests will be executed in both the client. This is how you can end-up with tests running 4 times (there is a separate issue that tests are executed twice and 2*2 = 4 !).

I'll try to add a way to reproduce on next failure.

**Comment by vicb** [comment]
just happened with the dart test, similar issue: the browser does not get closed

```
[13:48:03] Finished '!test.unit.dart/karma-server' after 932 ms
[13:48:03] Starting '!test.unit.dart/karma-run'...
INFO [karma]: Karma v0.12.31 server started at http://localhost:9877/
INFO [launcher]: Starting browser Dartium
Error: watch ENOSPC
    at exports._errnoException (util.js:746:11)
    at FSWatcher.start (fs.js:1172:11)
    at Object.fs.watch (fs.js:1198:11)
    at createFsWatchInstance (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:37:15)
    at setFsWatchListener (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:80:19)
    at EventEmitter.NodeFsHandler._watchWithNodeFs (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:228:14)
    at EventEmitter.NodeFsHandler._handleDir (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:403:23)
    at EventEmitter.<anonymous> (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:450:19)
    at EventEmitter.<anonymous> (/home/victor/dart/angular/node_modules/gulp-watch/node_modules/chokidar/lib/nodefs-handler.js:455:16)
    at FSReqWrap.oncomplete (fs.js:95:15)
```

**Comment by vicb** [comment]
and the repro in JS:

```
[15:50:12] Finished '!broccoli.js.dev' after 1.8 s
[15:50:12] Starting '!test.unit.js/karma-run'...
Error: ENOENT, stat '/home/victor/dart/angular/modules/angular2/src/test_lib/fake_async.es6___jb_bak___'
    at Error (native)
```

does not close the browser

**Comment by IgorMinar** [comment]
fixed by f9f917bfa49ea841a2bb2e977ccc48181fdbb63f

**Comment by vicb** [comment]
:+1:
